native/cocos/3d/models/SkinningModel.cpp

```diff
@@ -216,18 +249,105 @@ void SkinningModel::ensureEnoughBuffers(index_t count) {
             _buffers[i] = _device->createBuffer({
                 gfx::BufferUsageBit::UNIFORM | gfx::BufferUsageBit::TRANSFER_DST,
                 gfx::MemoryUsageBit::HOST | gfx::MemoryUsageBit::DEVICE,
-                pipeline::UBOSkinning::SIZE,
-                pipeline::UBOSkinning::SIZE,
+                pipeline::UBOSkinning::size,
+                pipeline::UBOSkinning::size,
             });
         }
 
         if (i >= _dataArray.size()) {
+            for (auto &data : _dataArray) {
+                delete[] data;
+            }
             _dataArray.resize(i + 1);
         }
 
         if (_dataArray[i] == nullptr) {
-            _dataArray[i] = ccnew ccstd::array<float, pipeline::UBOSkinning::COUNT>;
+            _dataArray[i] = new float[pipeline::UBOSkinning::count];
+            memset(_dataArray[i], 0, sizeof(float) * pipeline::UBOSkinning::count);
         }
     }
 }
+
+void SkinningModel::initRealTimeJointTexture(index_t jointCount) {
+    if (_realTimeJointTexture) {
+        delete _realTimeJointTexture;
+        _realTimeJointTexture = nullptr;
+    }
+    if (pipeline::SkinningJointCapacity::jointUniformCapacity < jointCount) {
+        _realTimeTextureMode = true;
+    }
+    if (!_realTimeTextureMode) return;
+    _realTimeJointTexture = new RealTimeJointTexture;
+    auto *device = gfx::Device::getInstance();
+    uint32_t texWidth = RealTimeJointTexture::WIDTH;
+    uint32_t texHeight = RealTimeJointTexture::HEIGHT;
+    gfx::Format textureFormat = gfx::Format::RGBA32F;
+
+    gfx::FormatFeature formatFeature = device->getFormatFeatures(gfx::Format::RGBA32F);
+    if (!(formatFeature & gfx::FormatFeature::SAMPLED_TEXTURE)) {
+        textureFormat = gfx::Format::RGBA8;
+        texWidth = 4 * RealTimeJointTexture::WIDTH;
+    }
+    uint32_t length = 4 * RealTimeJointTexture::WIDTH * RealTimeJointTexture::HEIGHT;
+    const size_t count = _dataArray.size();
+    for (size_t i = 0; i < count; i++) {
+        delete[] _dataArray[i];
```

Comment:
Just use delete, as `_dataArray[i]` is a pointer which points to `ccstd::array<> *`.